contracts/src/L1/rollup/ScrollChain.sol

```diff
@@ -48,15 +48,18 @@ contract ScrollChain is OwnableUpgradeable, PausableUpgradeable, IScrollChain {
     /// @notice The chain id of the corresponding layer 2 chain.
     uint64 public immutable layer2ChainId;
 
+    /// @notice The address of L1MessageQueue contract.
+    address public immutable messageQueue;
+
     /*************
      * Variables *
      *************/
 
     /// @notice The maximum number of transactions allowed in each chunk.
     uint256 public maxNumTxInChunk;
 
-    /// @notice The address of L1MessageQueue.
-    address public messageQueue;
+    /// @dev The storage slot used as L1MessageQueue contract, which is deprecated now.
+    address private __messageQueue;
 
     /// @notice The address of RollupVerifier.
     address public verifier;
```

Comment:
Should we also make `verifier` immutable? This won't change often.

Reply:
Yeah, we can.